### DISCUSSION TITLE
C# SequentialPlanner parameter chaining

### DISCUSSION THREAD

**Discussion by Ricochet73** [microsoft/semantic-kernel#2012] [Q&A]
I am testing implementing semantic functions in C# and how the SequentialPlanner produces plans. For this I have the following:
Func 1: public string GetValA(string code, string date)
Func 2: public string GetValB(string code, string date)
Func 3: GetCombinedVal(string code, string valA, string valB)

The ask then becomes - Give me the combined value for both codes "P1" and "P2" for date "X".

The planner correctly devises a plan to first call GetValA, then GelValB then GetCombinedVal for each of "P1" and "P2". However, what it does wrong is for Func 3 above  GetCombinedVal(string code, string valA, string valB), it will pass in the correct params for valA and valB (i.e both for P1 or both for P2), but it will not pass in the param "code". It matches the values for the correct code in the plan, it simply does not pass the code parameter to the final function. Is there a way to enforce this? Or am I doing something wrong?

**Comment by evchaki** [comment]
@Ricochet73  check out the new planner updates - https://devblogs.microsoft.com/semantic-kernel/migrating-from-the-sequential-and-stepwise-planners-to-the-new-handlebars-and-stepwise-planner/
